internal/abci/accumulator.go

```diff
@@ -253,12 +267,33 @@ func (app *Accumulator) InitChain(req abci.RequestInitChain) abci.ResponseInitCh
 func (app *Accumulator) BeginBlock(req abci.RequestBeginBlock) abci.ResponseBeginBlock {
 	defer app.recover(nil)
 
+	var ret abci.ResponseBeginBlock
+
 	//Identify the leader for this block, if we are the proposer... then we are the leader.
-	app.chain.BeginBlock(BeginBlockRequest{
+	r, err := app.chain.BeginBlock(BeginBlockRequest{
 		IsLeader: bytes.Equal(app.address.Bytes(), req.Header.GetProposerAddress()),
 		Height:   req.Header.Height,
 		Time:     req.Header.Time,
 	})
+	if err != nil {
+		app.fatal(err)
+		return ret
+	}
+
+	if false {
+		// Convert the synth TXN refs into events.
+		for _, syn := range r.SynthTxns {
```

Comment:
is this used for debugging?

Reply:
I'll remove it. A while ago, I updated `DeliverTx` to produce Tendermint events whenever a synthetic transaction was produced. Due to the changes in this PR, that event production needed to move to `BeginBlock`. However, returning events here makes one of the tests panic in the middle of Tendermint. So I disabled it. But I don't think anyone is using those events, so I'll just remove it.